### PR TITLE
Fix type checking for checked errors vs return type

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
$title.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/19299
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/19150

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by MaryamZi** [comment]
@chamil321 @ldclakmal @wggihan, can you please review the std lib changes? 

The relevant commits are e3a62a5 and ce6c789.

File: tests/jballerina-integration-test/src/test/resources/auth/src/authservices/17_auth_with_custom_providers_handlers.bal [hunk 1]
```diff
@@ -174,9 +174,13 @@ public type OutboundCustomAuthHandler object {
...
     public function prepare(http:Request req) returns http:Request|http:AuthenticationError {
-        string token = check self.authProvider.generateToken();
-        req.setHeader(http:AUTH_HEADER, "Custom " + token);
-        return req;
+        string|auth:Error token = self.authProvider.generateToken();
+        if (token is string) {
+            req.setHeader(http:AUTH_HEADER, "Custom " + token);
+            return req;
+        } else {
+            return http:AuthenticationError(message = token.reason(), cause = token);
```
Comment (ldclakmal) [inline tests/jballerina-integration-test/src/test/resources/auth/src/authservices/17_auth_with_custom_providers_handlers.bal]: return http:prepareAuthenticationError(token.reason(), token);

File: tests/jballerina-integration-test/src/test/resources/auth/src/authservices/17_auth_with_custom_providers_handlers.bal [hunk 2]
```diff
@@ -186,12 +190,13 @@ public type OutboundCustomAuthHandler object {
             string[] headerValues = resp.getHeaders(header);
             headerMap[header] = headerValues;
         }
-        string? token = check self.authProvider.inspect(headerMap);
+        string|auth:Error? token = self.authProvider.inspect(headerMap);
         if (token is string) {
             req.setHeader(http:AUTH_HEADER, "Custom " + token);
             return req;
+        } else if (token is auth:Error) {
+            return http:AuthenticationError(message = token.reason(), cause = token);
```
Comment (ldclakmal) [inline tests/jballerina-integration-test/src/test/resources/auth/src/authservices/17_auth_with_custom_providers_handlers.bal]:
```suggestion
            return http:prepareAuthenticationError(token.reason(), token);
```

File: tests/jballerina-integration-test/src/test/resources/auth/src/authservices/17_auth_with_custom_providers_handlers.bal (hunk 1 above)
Comment (hasithaa) [reply tests/jballerina-integration-test/src/test/resources/auth/src/authservices/17_auth_with_custom_providers_handlers.bal]: +1

**Pull request merged**